gapis/api/vulkan/replay.go

```diff
@@ -895,12 +895,55 @@ func getInitialCmds(ctx context.Context,
 	return []api.Cmd{}
 }
 
-// Melih TODO: This function may change as we moved other queries
-// Depending on how much code repetation it has and if it is avoidable
-// without making things complicated again
+func replayProfile(ctx context.Context,
+	intent replay.Intent,
+	dependentPayload string,
+	rrs []replay.RequestAndResult,
+	c *capture.GraphicsCapture,
+	device *device.Instance,
+	out transform2.Writer) error {
+
+	var layerName string
+	if device.GetConfiguration().GetPerfettoCapability().GetGpuProfiling().GetHasRenderStageProducerLayer() {
+		layerName = "VkRenderStagesProducer"
+	}
+
+	initialCmds := getInitialCmds(ctx, dependentPayload, intent, out)
+
+	transforms := make([]transform2.Transform, 0)
+	transforms = append(transforms, getCommonInitializationTransforms("ProfileReplay")...)
+
+	var request profileRequest
+	profileTransform := newEndOfReplay()
+
+	//Melih TODO: Remove this loop if we can guarantee if there is only one request
+	for _, rr := range rrs {
+		profileTransform.AddResult(rr.Result)
+		request = rr.Request.(profileRequest)
```

Comment:
The thing is, I don't think that it ever runs multiple profile requests at once. I want to clear it with Adam or Chris.

Reply:
Either the loop should be removed or the loop should do the right thing when iterated more than once. No point having code that we know does not work!

Reply:
Well, the comment is a question to other people who reviews to comment on if we are sure about this or not.

Reply:
So it is WRONG to try and force this in the UI.
What you CAN do is prevent multiple profile requests from getting batched together, Then evne if 2 requests come in, only one will happen at a time. However in that case you should probably panic if we ever end up with more than one (since something went terribly wrong).

Reply:
I have talked with Pascal and we don't want any changes in batching. Therefore, I copied the previous behaviour in the old replay to the new one and updated the PR.

Reply:
I still think we should not have code that we know does not work. If the loop does not behave correctly when iterated more than once it should be removed or the behaviour fixed. If Andrew is correct and the loop iterating more than once is an error, replace the loop with a "if iterations > 1 { panic() }"

Reply:
Pascal does not want to change the logic and advised to keep the same behaviour. I can try to fix this but it is going to a rabbit hole. If we have a complain about this situation which I have, it should be part of another PR after we discuss and decide what to do.